src/main.ts

```diff
@@ -0,0 +1,250 @@
+  stdout,
+  stderr,
+  config: { cachedRepositoriesDirectoryPath, defaultProjectNames },
+}: {
+  argv: string[];
+  stdout: SimpleWriteStream;
+  stderr: SimpleWriteStream;
+  config: {
+    cachedRepositoriesDirectoryPath: string;
+    defaultProjectNames: string[];
+  };
+}) {
+  const outputLogger = new OutputLogger({ stdout, stderr });
+  const workingDirectoryPath = WORKING_DIRECTORY_PATH;
+
+  const inputs = await parseInputs({ argv, outputLogger, defaultProjectNames });
+  /* istanbul ignore next: At the moment, there is no real way that Yargs could fail */
+  if (!inputs) {
+    process.exitCode = 1;
+    return;
+  }
+  const { templateRepositoryName, projectReferences } = inputs;
+
+  const template = await establishMetaMaskRepository({
+    repositoryReference: templateRepositoryName,
+    workingDirectoryPath,
+    cachedRepositoriesDirectoryPath,
+    outputLogger,
+  });
+
+  await lintProjects({
+    projectReferences,
+    template,
+    workingDirectoryPath,
+    cachedRepositoriesDirectoryPath,
+    outputLogger,
+  });
+}
+
+/**
+ * Parses command-line arguments and establishes parameters for the remainder of
+ * this tool.
+ *
+ * @param args - The arguments to this function.
+ * @param args.argv - The name of this executable and its arguments (as obtained
+ * via `process.argv`).
+ * @param args.outputLogger - Writable streams for output messages.
+ * @param args.defaultProjectNames - The set of MetaMask repositories
+ * that will be linted.
+ * @returns The parsed command-line arguments if parsing was successful, or null
+ * otherwise.
+ */
+async function parseInputs({
+  argv,
+  outputLogger,
+  defaultProjectNames,
+}: {
+  argv: string[];
+  outputLogger: OutputLogger;
+  defaultProjectNames: string[];
+}): Promise<Inputs | null> {
+  const args = await parseCommandLineArguments({ argv, outputLogger });
+
+  if (args) {
+    const projectReferences = args._.length > 0 ? args._ : defaultProjectNames;
+    return {
+      templateRepositoryName: DEFAULT_TEMPLATE_REPOSITORY_NAME,
+      projectReferences,
+    };
+  }
+
+  /* istanbul ignore next: At the moment, there is no real way that Yargs could fail */
+  return null;
+}
+
+/**
+ * Parses the command line using `yargs`.
+ *
+ * @param args - The arguments to this function.
+ * @param args.argv - The name of this executable and its arguments (as obtained
+ * via `process.argv`).
+ * @param args.outputLogger - Writable streams for output messages.
+ */
+async function parseCommandLineArguments({
+  argv,
+  outputLogger,
+}: {
+  argv: string[];
+  outputLogger: OutputLogger;
+}) {
+  let yargsFailure: { message: string; error: Error } | null = null;
+  /* istanbul ignore next: At the moment, there is no real way that Yargs could fail */
+  const onFail = (message: string, error: Error) => {
+    if (error) {
+      throw error;
+    }
+    yargsFailure = { message, error };
+  };
+
+  const yargs = createYargs(hideBin(argv))
+    .usage(USAGE_TEXT)
+    .help(false)
```

Comment:
Why disable `--help`? With this disabled, how is it possible for `options.help` below to be set?

Reply:
In order to write tests at the functional level, I need to control what gets output to stderr. I could mock `process.stderr` in the tests, but I don't want to completely shut out all output, because sometimes we like to add logging to help with debugging, and I thought that would have been annoying if that didn't work out of the box. To fix this, I disable Yargs's default `help` behavior and handle it myself. This still happens to work because I'm not using strict mode — I can't, otherwise I wouldn't be able to accept arbitrary arguments — so Yargs just puts unrecognized options in `argv`, and I can access them there.

That said, 1) even if I can't use strict mode, I can use strict _options_ mode, and 2) I've just noticed that setting `help(false)` removes `--help` from the usage output, so I should put it back.